sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1347,20 +1350,27 @@ class Analyzer(
 
     def resolveAssignments(
         assignments: Seq[Assignment],
-        mergeInto: MergeIntoTable): Seq[Assignment] = {
+        mergeInto: MergeIntoTable,
+        resolveValueWithBothTables: Boolean): Seq[Assignment] = {
```

Comment:
This name is a little ambiguous. What does `resolveValueWithBothTables = false` mean? which table will it use, source or target? Maybe better name is `resolveValuesWithSourceOnly`.